APE12.rst

```diff
@@ -0,0 +1,188 @@
+    drwxr-xr-x  5 stuart users 4.0K Nov 17 14:41 ./
+    drwx------ 69 stuart users 4.0K Nov 17 14:41 ../
+    -rw-r--r--  1 stuart users  36K Nov 17 14:41 ah_bootstrap.py
+    -rw-r--r--  1 stuart users 1.8K Nov 17 14:41 appveyor.yml
+    drwxr-xr-x  6 stuart users 4.0K Nov 17 14:41 cadairpy/
+    drwxr-xr-x  4 stuart users 4.0K Nov 17 14:41 docs/
+    -rw-r--r--  1 stuart users  13K Nov 17 14:41 ez_setup.py
+    -rw-r--r--  1 stuart users  655 Nov 17 14:41 .gitignore
+    -rw-r--r--  1 stuart users  108 Nov 17 14:41 .gitmodules
+    drwxr-xr-x  2 stuart users 4.0K Nov 17 14:41 licenses/
+    -rw-r--r--  1 stuart users 1.1K Nov 17 14:41 MANIFEST.in
+    -rw-r--r--  1 stuart users  742 Nov 17 14:41 README.md
+    -rw-r--r--  1 stuart users   71 Nov 17 14:41 readthedocs.yml
+    -rw-r--r--  1 stuart users  160 Nov 17 14:41 .rtd-environment.yml
+    -rw-r--r--  1 stuart users 1.2K Nov 17 14:41 setup.cfg
+    -rwxr-xr-x  1 stuart users 4.0K Nov 17 14:41 setup.py*
+    -rw-r--r--  1 stuart users 4.8K Nov 17 14:41 .travis.yml
+
+
+
+
+Branches and pull requests
+--------------------------
+
+A version of the package template converted to a Cookiecutter project is `here
+<https://github.com/Cadair/package-template/tree/cookiecutter>`_. It can be
+tested by installing the cookiecutter package from pip or conda-forge and
+running::
+
+  cookiecuttter -c cookiecutter gh:cadair/package-template
+
+
+Implementation
+--------------
+
+The changes to the package template are to move the template into a subdirectory
+and to change all occurrences of ``packagename`` and other user modifiable
+variables to Jinja template placeholders such as ``{{ cookiecutter.package_name }}``.
+
+A draft implementation is `here <https://github.com/Cadair/package-template/tree/cookiecutter>`_. Which is in
+`PR #202 <https://github.com/astropy/package-template/pull/202>`_ to the
+package-template repository.
+
+As well as the changes to the package template the documentation in the Astropy
+docs regarding the package template will be significantly simplified.
+
+
+Backward compatibility
+----------------------
+
+As implemented in
+`Cadair/package-template <https://github.com/Cadair/package-template/tree/cookiecutter>`_
+users who have followed the instructions where they use the package template as
+a git remote will no longer be able to pull updates from the
+``package_template`` git repository. This could be mitigated by automatically
+generating a rendered version of the package template in another git repository
+or in a different branch.
+
+
+Alternatives
+------------
+
+Keeping the current solution. This prevents any trouble for anyone using the
+'git update' method at the moment, but it does not provide any of the templating
+benefits presented above.
+
+Cookiecutter is not the only project templating system, others such as
+`diecutter <https://diecutter.readthedocs.io/>`_ could be used. Cookiecutter was
+chosen due to wide adoption and implementation in Python.
```

Comment:
Another alternative is to write our own setup script. Might be worth mentioning at least.